cdap-gateway/src/test/java/co/cask/cdap/gateway/handlers/metrics/MetricsHandlerTestRun.java

```diff
@@ -195,10 +195,22 @@ public void testSearchWithTags() throws Exception {
 
     // verify "*"
 
+    // components (services) are listed as optional, as metrics are cleared between tests,
+    // some of the services might not have emitter metrics yet since this test begining,
```

Comment:
since the beginning of this test. as we clear metrics in after method in the test base.